sdk/tests/conformance2/textures/misc/copy-texture-image-webgl-specific.html

```diff
@@ -130,13 +130,58 @@
     gl.deleteTexture(texture);
 }
 
+function copytexsubimage3d_texture_wrongly_initialized() {
+    debug("");
+    debug("Testing copytexsubimage3d_texture_wrongly_initialized");
+    var texture = [];
+    texture[0] = gl.createTexture();
+    texture[1] = gl.createTexture();
+    var layer = 0;
+    var width = 2;
+    var height = 2;
+    var depth = 2;
+    gl.bindTexture(gl.TEXTURE_2D, texture[0]);
+    var uint = new Uint8Array([0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F, 0x10]);
+    gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGBA, width, height, 0, gl.RGBA, gl.UNSIGNED_BYTE, uint);
+
+    var fbo = gl.createFramebuffer();
+    gl.bindFramebuffer(gl.FRAMEBUFFER, fbo);
+    gl.framebufferTexture2D(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.TEXTURE_2D, texture[0], 0);
+    if (gl.checkFramebufferStatus(gl.FRAMEBUFFER) == gl.FRAMEBUFFER_COMPLETE) {
+        gl.bindTexture(gl.TEXTURE_3D, texture[1]);
+        gl.texStorage3D(gl.TEXTURE_3D, 1, gl.RGBA8, width, height, depth);
+        gl.copyTexSubImage3D(gl.TEXTURE_3D, 0, 0, 0, layer, 0, 0, width, height);
+        gl.framebufferTextureLayer(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, texture[1], 0, layer);
```

Comment:
Ken and Zhenyao, thanks for your explanation. But I still confused by the Spec and the implementation in Chromium. Let me state it clearly. 

As you said, the texStorage3D doesn't initiate the texture, CopyTexSubImage3D should do this. This feature is done by my patch at https://codereview.chromium.org/2232743002/. But that patch has not been merged yet. However, when the test reads pixels (on line 156) from the texture, it found that the texture has been initialized to 0 (So the test fails in latest Chromium). So I think it is framebufferTextureLayer do this initialization work. 
But the WebGL spec doesn't say that framebufferTextureLayer should clear the uninitialized texture.  

However, the spec about TexStorage3D in WebGL 2 says that: "The image contents are set as if a buffer of sufficient size initialized to 0 would be passed to each texImage3D call in the pseudocode in The OpenGL ES 3.0 specification section 3.8.4", So I think it should be initialized to 0 at the very beginning texStorage3D allocate the memory for texture. This is webgl specific, because in gles, it just allocate the memory, but don't initialized the memory. 

So It seems to me that the implementation about TexStorage3D and framebufferTextureLayer conflicts with the WebGL 2 spec about these two APIs(Maybe I misunderstand you and the code).